src/app/plugins/scores.plugin.ts

```diff
@@ -3,123 +3,146 @@ import { Plugin } from '../../common/plugin';
 import Constants from '../../common/constants';
 import { IContainer, IMessage, ChannelType } from '../../common/types';
 
-
 export class ScoresPlugin extends Plugin {
   public name: string = 'NCAA Scores Plugin';
   public description: string = 'Gets score of a football game.';
   public usage: string = 'scores <team name>; ex scores UCF';
   public permission: ChannelType = ChannelType.Public;
   public pluginChannelName: string = Constants.Channels.Public.Sports;
 
+  private endpoint: string = 'http://www.espn.com/college-football/bottomline/scores';
+  private maxNumDisplay: number = 3;
+
   constructor(public container: IContainer) {
     super();
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const endpoint = 'http://www.espn.com/college-football/bottomline/scores';
     let desiredTeamQuery = this._parseTeam(args || []);
 
     //If no args, team is UCF by default
     if (desiredTeamQuery === '') {
       desiredTeamQuery = 'UCF';
     }
 
+    const visitorDataRegex: RegExp = /[=\^][a-zA-Z]+%20([a-zA-Z]*%20)?[0-9]+/; //Isolates visiting team's data
+    const homeDataRegex: RegExp = /(?<=%20%20)\^?(\(\d+\))?[a-zA-Z(%20)]+%20[0-9]+/; //Isolates home team's data
 
     try {
-      const response = await this.container.httpService.get(endpoint);
+      const response = await this.container.httpService.get(this.endpoint);
       const games = response.data.split('?'); //Each game ends with a ?
 
-
+      let messagesSent = 0;
       let teamFound = false;
       for (const game of games) {
         if (this._containsAllTokens(game, desiredTeamQuery.split(' '))) {
```

Comment:
nit: rather than putting everything in this big `if`, check the negative of the condition first, and skip the `game` if necessary.

Reply:
so check `if(!this._containsAllTokens(...)) {break;}`?

Reply:
`continue` but yes